test/distributed/_tensor/test_math_ops.py

```diff
@@ -361,9 +362,11 @@ def _replicate_fn(name, module, device_mesh):
             with comm_mode:
                 y_dist.sum().backward()
 
-            self.assertLessEqual(
+            expected_bwd_comm = 0 if shard_dim < norm_idx else 1
```

Comment:
So here the expected communications for forward and backward use the same criteria. I'm trying to understand what is going on. Please help correct me if I'm wrong.

I'm able to parse the forward pass. If `shard_dim < norm_idx`, everything needed is at hand when layer norm is called; o/w we need to all-gather sharded data from other ranks to perform the layer norm. So the code aligns with my interpretation.

The backward is a bit difficult for me. If `shard_dim < norm_idx`, that means no all-gather in the forward pass, and the backward locally updates the weight and bias. But the weight and bias have only seen sharded data, thus they need to be all-reduced among all ranks -- which would mean `expected_bwd_comm = 2`.
When `shard_dim >= norm_idx`, an all-gather happened in the forward pass, so in the backward pass every rank have updated their weight and bias on all data. Thus no communication is needed.

Based on my understanding, we should have something like `expected_bwd_comm = 2 if shard_dim < norm_idx else 0`.